docs/GuidGuidRecommender.md

```diff
@@ -0,0 +1,144 @@
+    based on the coinstallation data.
+3. Recommendations for a given add-on are [selected](#selecting-recommendations)
+    by retaining the N add-ons in the candidate list
+    with the highest relevance scores.
+
+Once the recommender has been trained,
+we apply various [metrics](./GraphRepresentation.md#quality-and-health-metrics)
+to the coinstallation data, candidate lists, and recommendations,
+to evaluate the overall health of the system
+and the quality of recommendations.
+
+The recommender system is implemented by the [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10) class,
+which is instantiated by the [TaarLiteAppResource](../taar_lite/app/production.py#L44).
+
+Further technical details are provided on [this page](./GraphRepresentation.md).
+
+
+# Input data
+
+The primary data source used in generating recommendations
+is the file `guid_coinstallation.json`,
+which is generated by the ETL jobs described in the [README](../README.md).
+It contains coinstallation counts for each pair of add-ons
+belonging to a filtered list.
+
+Coinstallation counts are computed based on the Telemetry data
+submitted by a random sample of recently active Firefox profiles.
+For each pair of add-ons in the list,
+we count the number of profiles in which both add-ons are installed and enabled.
+
+The data is stored in "sparse matrix representation" as a nested object/dict,
+in which each add-on GUID maps to a further object
+mapping coinstalled GUIDS to their coinstallation count,
+dropping zero counts.
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+it is supplied as `raw_coinstall_dict`.
+
+Using this dataset as the basis for recommendations
+entails the following assumptions:
+
+- The set of add-ons appearing in the coinstallation dataset represent
+    the universe of known add-ons for the purpose of generating recommendations.
+- We are only dealing with add-ons that were coinstalled
+    with at least one other add-on on the list
+    (otherwise they would not appear in the file at all).
+- Profiles with only a single add-on installed do not contribute to the counts.
+
+__TODO:__ How are system add-ons handled? Add a note about this.
+
+In computing relevance scores, we also draw on an auxiliary dataset
+containing an overall ranking for each add-on in our list,
+read from the file `guid_install_ranking.json`.
+This is used to break ties when sorting candidates on relevance score,
+and in pruning out rare add-ons.
+It is currently computed from overall add-on installation counts
+obtained from AMO.
+
+This data is stored as an object/dict mapping add-on GUIDs to rank scores.
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+it is supplied as `tie_breaker_dict`.
+It is also used as the `ranking_dict` passed to the [`MinInstallPrune`](../taar_lite/recommenders/treatments.py#L36) treatment.
+
```

Comment:
Again, separate out to the general Object related concerns only. 

Also, it's probably worth a note explaining why the ranking dict cannot be computed from the co-install dict. Just to be extra clear. (Answer being ranking dict lets us get the popularity of addons that are only installed by one client, and that's it I think, maybe there's more)

Reply:
It's perhaps worth noting that I deliberately called the recommender attribute `tie_breaker_dict` because anything could be used to break ties. 

The fact that the same data is used as an input to `MinInstallPrune` is coincidental and I wanted to capture that abstraction.

Reply:
When tie_breaking aka ordering becomes just another treatment, that will be extra clear, and there will be no `tie_breaker_dict` input to the recommender, it'll just be an input to another treatment, so I would suggest removing these last two paragraphs as they'll go soon anyway and we'll add a "Deterministic Ordering" section to the Treatments page.

Reply:
I've updated these paragraphs to address your points. I think I'll leave them in for now, as they're already there, and remove them as a part of the PR that converts tie breaking to a treatment.